mentoring/dashboard_visual.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2014-2015 Harvard, edX & OpenCraft
+#
+# This software's license gives you freedom; you can copy, convey,
+# propagate, redistribute and/or modify this program under the terms of
+# the GNU Affero General Public License (AGPL) as published by the Free
+# Software Foundation (FSF), either version 3 of the License, or (at your
+# option) any later version of the AGPL published by the FSF.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+# General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program in a file in the toplevel directory called
+# "AGPLv3".  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Visual Representation of Dashboard State.
+
+Consists of a series of images, layered on top of each other, where the appearance of each layer
+can be tied to the average value of the student's response to a particular Problem Builder
+block.
+
+For example, each layer can have its color turn green if the student's average value on MCQs in
+a specific Problem Builder block was at least 3.
+"""
+
+
+class DashboardVisualData(object):
+    """
+    Data about the visual representation of a dashboard.
+    """
+    def __init__(self, blocks, rules, color_for_value):
+        """
+        Construct the data required for the optional visual representation of the dashboard.
+
+        Data format accepted for rules is like:
+        {
+            "images": [
+                "step1.png",
```

Comment:
Perhaps these should say `"/studio/step1.png"`, etc. That's how it's likely to be in practice.